cypress/integration/mirror_spec.js

```diff
@@ -1,11 +1,111 @@
+    });
+  });
+
+  describe("2. String Uppercaser", () => {
+    it("uppercases the entered string", () => {
+      cy.get("#uppercaser-input")
+        .type("Hello world!")
+        .get("#uppercaser-form button")
+        .click()
+        .get("#uppercaser-output")
+        .contains("HELLO WORLD!");
+    });
+  });
+
+  describe("3. Palindrome Detector", () => {
+    it("displays true when the string is a palindrome", () => {
+      cy.get("#palindrome-input")
+        .type("tacocat")
+        .get("#palindrome-form button")
+        .click()
+        .get("#palindrome-output")
+        .contains("It is true that tacocat is a palindrome");
+    });
+
+    it("displays false when the string is not a palindrome", () => {
+      cy.get("#palindrome-input")
+        .type("nachodog")
+        .get("#palindrome-form button")
+        .click()
+        .get("#palindrome-output")
+        .contains("It is false that nachodog is a palindrome");
+    });
+  });
+
+  describe("4. Even Checker", () => {
+    it("displays true when the number is even", () => {
+      cy.get("#even-checker-input")
+        .type("42")
+        .get("#even-checker-form button")
+        .click()
+        .get("#even-checker-output")
+        .contains("It is true that 42 is even");
+    });
+
+    it("displays false when the number is even", () => {
+      cy.get("#even-checker-input")
+        .type("117")
+        .get("#even-checker-form button")
+        .click()
+        .get("#even-checker-output")
+        .contains("It is false that 117 is even");
+    });
+  });
+
+  describe("5. Number Doubler", () => {
+    it("displays the number doubled", () => {
+      cy.get("#doubler-input")
+        .type("12")
+        .get("#doubler-form button")
+        .click()
+        .get("#doubler-output")
+        .contains("12 doubled is 23");
+    });
+  });
+
+  describe("6. Average of Three Numbers", () => {
+    it("displays the average", () => {
+      cy.get("#average-input-1")
+        .type("10")
+        .get("#average-input-2")
+        .type("35")
+        .get("#average-input-3")
+        .type("45")
+        .get("#average-form button")
+        .click()
+        .get("#average-output")
+        .contains("The average of 10, 35, and 45 is 30");
+    });
+  });
+
+  describe("Bonus: Vowel Remover", () => {
+    it("removes vowels excluding y when the checkbox is not checked", () => {
+      cy.get("vowel-remover-input")
+        .type("happy days ahead")
+        .get("#vowel-remover-form button")
+        .click()
+        .get("#vowel-remover-output")
+        .contains("hppy dys hd");
+    });
+
+    it("removes vowels including y when the checkbox is checked", () => {
+      cy.get("vowel-remover-input")
+        .type("happy days ahead")
+        .get("y-is-vowel-checkbox")
+        .click()
+        .get("#vowel-remover-form button")
+        .click()
+        .get("#vowel-remover-output")
+        .contains("hpp ds hd");
+    });
+  });
```

Comment:
I forgot to mention this! The bonus is not required so we don't want circle to show us a red x if they do not do the bonus. We have solved this in the past by moving the bonus spec to a separate file and modifying the test script in package.json to not run the bonus spec. Example: https://github.com/joinpursuit/functions-lab/blob/master/package.json#L13-L16

Reply:
I moved it to a `bonus_spec.js` alongside `mirror_spec.js`. The test command in the package is `npx cypress open` so I'll leave it to you if you want any more command edits. 😉

Reply:
Ooh this comment has a couple of good catches!
1. I don't think npx is installed on our students' machines
2. We will have to fiddle with the commands so that the bonus doesn't run in the automation